synapse/handlers/identity.py

```diff
@@ -159,11 +159,14 @@ def try_unbind_threepid(self, mxid, threepid):
             Deferred[bool]: True on success, otherwise False if the identity
             server doesn't support unbinding
         """
-        id_servers = yield self.store.get_id_servers_user_bound(
-            user_id=mxid,
-            medium=threepid["medium"],
-            address=threepid["address"],
-        )
+        if threepid.get("id_server"):
```

Comment:
why is this in the dict rather than a function param? (and anyway, the docstring needs updating)

Reply:
Broadly because this is how its handled in the `bind` path where the threepid dict contains the identity server. I don't mind pulling it out but then we probably want to also pull out `medium` and `address`